### DISCUSSION TITLE
TinyMCE in Chyrp Lite?

### DISCUSSION THREAD

**Discussion by JavaFenom** [xenocrat/chyrp-lite#240] [General]
Hello! How can I run TinyMCE in Chyrp Lite? The CKEditor module from 2017, which **xenocrat** brought up in the discussion, doesn't work. Please help, who knows how to run TinyMСЕ into Chyrp Lite.

**Comment by xenocrat** [comment]
Hello there,

You would implement this with a [basic module](https://github.com/xenocrat/chyrp-lite/wiki/Making-Your-First-Module) that responds to to the "admin_head" and "admin_javascript" [trigger calls](https://github.com/xenocrat/chyrp-lite/wiki/Introduction-to-Triggers). I guess you'd also want to hide Chyrp Lite's own toolbar and tray. I've attached a very basic module to do all that. You'll need to modify _tinymce.php_ to insert a valid TinyMCE API key where it currently has `'no-api-key'`.

I used the [TinyMCE quick start guide](https://www.tiny.cloud/docs/tinymce/6/cloud-quick-start/) to create this module; please refer to it for other questions specific to TinyMCE.

A very basic TinyMCE module: [tinymce.zip](https://github.com/xenocrat/chyrp-lite/files/14234154/tinymce.zip)

**Reply by JavaFenom** [reply]
Thank you Xenocrat!

**Reply by JavaFenom** [reply]
Hello xenocrat! Creating "Page" and "Post" does not work, because the **"Publish" and "Save" buttons** do not work (with you TinyMCE module). What to do and how to fix this?

**Reply by xenocrat** [reply]
This is one of many reasons I hate these WYSIWYG editors. It's quite ridiculous that a simple writeback to the replaced DOM element isn't automatic on blur, and also such a bloody pain to implement manually. I've updated the bundle (see below).

In order to test the functionality, I had to download the "community" edition of TinyMCE. So now in _tinymce.php_ you have the option to return `$cloud` for the cloud version, or `$hosted` for the version downloaded and added to the module. You're on your own from here on.

Download: [tinymce.zip](https://github.com/xenocrat/chyrp-lite/files/14298189/tinymce.zip)

**Reply by JavaFenom** [reply]
Thank you Daniel! Why you don't use Editor.md (Open Source Markdown editor)? https://github.com/pandao/editor.md

**Reply by xenocrat** [reply]
Thanks for the link - that's an interesting project - but I want to avoid adding dependencies to Chyrp Lite wherever possible.
